web3wagers/gostarkme
# add test_fund_deployed_event in test_fund_manager.cairo and add pub f…

…unc in fundManager.cairo

# Pull Request

- [ ] Closes #147 
- [x] Added tests (if necessary)
- [x] Run formatting
- [ ] Commented the code

## Changes description

This test function validates that the FundDeployed event is emitted correctly when a new fund is created in the fund manager contract.

Test Steps:

	1.	Contract Initialization: Gets the address of the fund manager contract (Fund Manager) and creates a dispatcher (fund_manager_contract) to interact with it.
	2.	Setting the Caller Address: Sets the caller address as the owner (OWNER()) using start_cheat_caller_address_global.
	3.	Event Capture: Initializes an event spy (spy_events()) to monitor and analyze emitted events.
	4.	Retrieving the Current ID: Fetches the current_id of the fund before deploying a new one.
	5.	Creating a New Fund: Calls newFund() to deploy a new fund with the values NAME() and GOAL().
	6.	Verifying Emitted Events: Uses spy.assert_emitted to confirm that the FundDeployed event was emitted correctly with the expected attributes: contract_address, fund_address, fund_id, and owner.
![Screenshot 2024-11-01 at 12 12 44 AM](https://github.com/user-attachments/assets/031c056b-56f9-4f73-848a-cc63d68b4b00)

## Current output
![Screenshot 2024-11-01 at 12 12 30 AM](https://github.com/user-attachments/assets/20a72b0f-48cb-4fbf-8888-5cea8bbcb60c)

## Time spent breakdown
3 days

## Comments

At first I had problems with the configuration but after solving the versions, everything flowed correctly
